streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -62,7 +57,7 @@
      * @throws ProcessorStateException if the state manager cannot be created
      */
     AbstractTask(final TaskId id,
-                 final Collection<TopicPartition> partitions,
+                 final Set<TopicPartition> partitions,
```

Comment:
Why reducing the type generality here?

Reply:
In `StreamTask` constructor we call `consumer.committed(...)` that takes a `Set<TopicPartitions>` -- hence, it's easier to pass in a `Set` from the beginning on. Second, thinking about it, a `Collection` is actually too generic anyway -- we would not want to assign a `List` of partitions that may contain duplicates IMHO.